### ISSUE TITLE
Kit edit bug???

### ISSUE THREAD

**Issue by AriaCraft2019** [FrozedClubDevelopment/FrostCommunity#51]
### Expected behavior

Items can be successfully changed.

### Observed/Actual behavior

/

### How to Recreate (if known)

Video
https://youtu.be/cSutUEhXLcQ

### Stack trace

https://pastebin.com/eY8HA585

### Plugin list

![Desktop Screenshot 2022 12 24 - 20 29 04 67](https://user-images.githubusercontent.com/82739427/209436362-569580e4-f27a-46e1-86dd-555b991fdb5b.png)
![Desktop Screenshot 2022 12 24 - 20 28 56 49](https://user-images.githubusercontent.com/82739427/209436364-74ef7504-850f-4a84-99c2-032a997ab738.png)

use CarbonSpigot 3.4

### Frost version

11.5

### Agreements

- [X] I have searched for and ensured there isn't already an open issue regarding this.

**Comment by Elb1to** [comment]
Incomplete stacktrace, though, that is caused by one of the anticheats you're using together (Intave and Matrix).

I'd suggest trying every check individually until you find the one that causes that issue to occur.

**Comment by AriaCraft2019** [comment]
We removed all AntiCheat and AntiCrash plugins. but it still has the problem.
![Desktop Screenshot 2023 01 02 - 15 38 44 64](https://user-images.githubusercontent.com/82739427/210205320-c344a8c4-eaeb-4f25-a444-06e005c2eb85.png)
[warn.txt](https://github.com/FrozedClubDevelopment/FrostCommunity/files/10330466/warn.txt)

Video（All AntiCheat and AntiCrash were uninstalled at the time the video was shot.）：https://www.youtube.com/watch?v=IJHS6-djCaQ

**Comment by Elb1to** [comment]
Then there’s another plugin causing it, since it doesn’t happen on any other server running Frost.

Try removing plugins one by one until you find the one that causes it.

**Comment by AriaCraft2019** [comment]
We've culled to this, but the problem persists, could this be a database or Spigot issue?
![Desktop Screenshot 2023 01 03 - 08 21 26 54](https://user-images.githubusercontent.com/82739427/210286908-d26d71de-37ef-4ed8-adac-3e6d56bea576.png)

**Comment by Elb1to** [comment]
Something is interfering with Frost, the stacktrace isn't complete so I can't tell what's causing it.

**Comment by AriaCraft2019** [comment]
[08:14:35] [WARN]: Exception in thread "Craft Scheduler Thread - 0" org.apache.commons.lang.UnhandledException: Plugin Frost v11.5 generated an exception while executing task 24
[08:14:35] [WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftAsyncTask.run(SourceFile:57)
[08:14:35] [WARN]: 	at org.github.paperspigot.ServerSchedulerReportingWrapper.run(SourceFile:22)
[08:14:35] [WARN]: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[08:14:35] [WARN]: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[08:14:35] [WARN]: 	at java.lang.Thread.run(Thread.java:748)
[08:14:35] [WARN]: Caused by: java.util.ConcurrentModificationException
[08:14:35] [WARN]: 	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:911)
[08:14:35] [WARN]: 	at java.util.ArrayList$Itr.next(ArrayList.java:861)
[08:14:35] [WARN]: 	at com.bizarrealex.aether.Aether$1.run(Aether.java:109)
[08:14:35] [WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftTask.run(SourceFile:59)
[08:14:35] [WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftAsyncTask.run(SourceFile:53)
[08:14:35] [WARN]: 	... 4 more
[08:14:53] [WARN]: Exception in thread "Craft Scheduler Thread - 6" org.apache.commons.lang.UnhandledException: Plugin Frost v11.5 generated an exception while executing task 108
[08:14:53] [WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftAsyncTask.run(SourceFile:57)
[08:14:53] [WARN]: 	at org.github.paperspigot.ServerSchedulerReportingWrapper.run(SourceFile:22)
[08:14:53] [WARN]: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[08:14:53] [WARN]: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[08:14:53] [WARN]: 	at java.lang.Thread.run(Thread.java:748)
[08:14:53] [WARN]: Caused by: net.minecraft.server.v1_8_R3.CancelledPacketHandleException


There is a longer warn here, and the rest of the warn are as short as the one above. There is a phenomenon that there is no problem to edit KIT for the first time after the server is opened, but if you save it and edit it again, there will be a problem.

https://www.youtube.com/watch?v=AeX8cmYGaYo

**Comment by Elb1to** [comment]
I need the full stacktrace for this one:

```
[08:14:53] [WARN]: Exception in thread "Craft Scheduler Thread - 6" org.apache.commons.lang.UnhandledException: Plugin Frost v11.5 generated an exception while executing task 108
[08:14:53] [WARN]: at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftAsyncTask.run(SourceFile:57)
[08:14:53] [WARN]: at org.github.paperspigot.ServerSchedulerReportingWrapper.run(SourceFile:22)
[08:14:53] [WARN]: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[08:14:53] [WARN]: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[08:14:53] [WARN]: at java.lang.Thread.run(Thread.java:748)
[08:14:53] [WARN]: Caused by: net.minecraft.server.v1_8_R3.CancelledPacketHandleException
```

The first stacktrace is unrelated, that's caused by the scoreboard.

Try running Frost alone with just the KnockbackController and see if it happens, if it does, then it's something from the Spigot you're using.

**Comment by AriaCraft2019** [comment]
It's really weird, we tried carbonspigot and ispigot, even regenerated paper.yml, spigot.yml, bukkit.yml files, reset the database, the problem still exists. plugin has been trimmed down to the bare minimum.
![Desktop Screenshot 2023 01 03 - 19 35 03 92](https://user-images.githubusercontent.com/82739427/210350158-3c4e6d7d-e03a-41e7-b856-80a125085f46.png)


[08:14:53] [WARN]: Exception in thread "Craft Scheduler Thread - 6" org.apache.commons.lang.UnhandledException: Plugin Frost v11.5 generated an exception while executing task 108
[08:14:53] [WARN]: at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftAsyncTask.run(SourceFile:57)
[08:14:53] [WARN]: at org.github.paperspigot.ServerSchedulerReportingWrapper.run(SourceFile:22)
[08:14:53] [WARN]: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[08:14:53] [WARN]: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[08:14:53] [WARN]: at java.lang.Thread.run(Thread.java:748)
[08:14:53] [WARN]: Caused by: net.minecraft.server.v1_8_R3.CancelledPacketHandleException

That's all that popped up on the server.


I also found that when this happens, it quits in the kitedit interface no matter which item bar is clicked.
https://youtu.be/f2U0BTv6SEM
